src/engine/renderer/tr_shader.cpp

```diff
@@ -760,6 +760,12 @@ static void ParseExpression( const char **text, expression_t *exp )
 
 			case opcode_t::OP_NUM:
 				op.value = atof( token );
+
+				if ( isColor && tr.worldLinearizeTexture )
```

Comment:
This doesn't make sense since expressions can contain any number of mathematical operations. The final result of evaluating the expression should be converted. You shouldn't convert each number that appears in the expression.

Reply:
Annoying… That's what I mentionned here:
https://github.com/DaemonEngine/Daemon/pull/1687#issuecomment-3060634276

In practice no one writes something else than lone numbers, but yeah, only render time conversion after evaluation is 100% safe, that's very annoying.

Reply:
I revamped that, and modified the expression evaluator to do conversions when needed. 

I did a much deeper revamp where the expression evaluator is aware of the returned value needing to be clamped or to be converted from sRGB.

Because those conversions are done at render time, I also implemented a cheap from-sRGB convertor which is used when `r_accurateSRGB` is false, in the same way a cheap to-sRGB convertor can be used in the `cameraEffect` GLSL shader.